tests/test_huave_morphology.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+from elotl.nahuatl.morphology import Analyser
+from json import JSONEncoder
+import json
+
+
+class TokenEncoder(JSONEncoder):
+	def default(self, o):
+		return o.__dict__
+
+class TestAnalysis(unittest.TestCase):
+	analyser = Analyser()
+
+	def test_analysis_tokenisation(self):
+
+		inp = "tambas."
+		outp = [{"wordform": "tambas", "analyses": [[[{"lemma": "mb", "pos": "VERB", "feats": {"Person": "1", "Tense": "Past", "Number": "Sing", "Subcat": "Intr"}}], 0.0]], "pos": "VERB", "lemma": "mb"}, {"wordform": ".", "analyses": [[[{"lemma": ".", "pos": "PUNCT", "feats": {}}], 0.0]], "pos": "PUNCT", "lemma": "."}]
+		res = self.analyser.analyse(inp, tokenise=True)
+		outpj = json.dumps(res, cls=TokenEncoder)
```

Comment:
I think this is supposed to be `outpj = json.dumps(outp, cls=TokenEncoder)` right?

Reply:
when i make this change the tests no longer pass :(

Reply:
The `outp` is already in JSON (python internal format). The `res` from the analyser is not in that format (it's some object type), so I use `json.dumps()` to serialise it before doing the assert comparison.

Reply:
Note that I'm happy to reorganise this if there is a better way! :)

Reply:
ah, okay then I think the issue is that the comparison is using `outpj` (which is identical to `resj`) instead of `output`